backend/integration_test/elasticsearch/count_service_test.go

```diff
@@ -32,27 +40,110 @@ func TestCount(t *testing.T) {
 		}
 		relevantCountInfo := countInfo["differentTime"]
 		assert.Equal(t, int64(len(logsOfDifferentTime)), relevantCountInfo.Occurrences)
 		assert.Equal(t, int64(numWithinBucket), relevantCountInfo.CoOccurrences)
-		err = deleteAllDocuments(es, "log_index")
+	})
+
+	t.Run("should store new entries into the database if nothing else is there", func(t *testing.T) {
+		err := deleteAllDocuments(es)
 		if err != nil {
 			t.Errorf("Failed to delete all documents: %v", err)
 		}
+		numWithinBucket := 4
+		initialTime := time.Date(2021, 1, 1, 0, 0, 0, 204, time.UTC)
+		newLog := model.LogEntry{
+			ClusterId: "initialClusterId",
+			Timestamp: initialTime,
+		}
+		logsOfDifferentTime := makeLogsOfSameClusterId("differentTime", initialTime.Add(time.Second), numWithinBucket)
+		err = loadLogsIntoElasticsearch(ac, logsOfDifferentTime)
+		if err != nil {
+			t.Error("Failed to load logs into elasticsearch")
+		}
+		err = loadLogsIntoElasticsearch(ac, []model.LogEntry{newLog})
+		if err != nil {
+			t.Error("Failed to load logs into elasticsearch")
+		}
+		buckets := []service.Bucket{2500}
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		err = countService.CountAndUpdateOccurrences(ctx, newLog, buckets)
+		if err != nil {
+			t.Errorf("Failed to count occurrences: %v", err)
+		}
+		searchQueryBody := countQuery(newLog.ClusterId)
+		docs, err := ac.Search(ctx, searchQueryBody, elasticsearch.CountIndexName, 100)
+		if err != nil {
+			t.Errorf("Failed to search for count: %v", err)
+		}
+		countEntries, err := convertCountDocsToCountEntries(docs)
+		if err != nil {
+			t.Errorf("Failed to convert count docs to count entries: %v", err)
+		}
+		countEntry := countEntries[0]
+		assert.Equal(t, int64(numWithinBucket), countEntry.Occurrences)
+		assert.Equal(t, int64(numWithinBucket), countEntry.CoOccurrences)
+	})
+
+	t.Run("should update existing entries in the database", func(t *testing.T) {
+		err := deleteAllDocuments(es)
+		if err != nil {
+			t.Errorf("Failed to delete all documents: %v", err)
+		}
+		numWithinBucket := 4
+		initialTime := time.Date(2021, 1, 1, 0, 0, 0, 204, time.UTC)
+		newLog := model.LogEntry{
+			ClusterId: "initialClusterId",
+			Timestamp: initialTime,
+		}
+		logsOfDifferentTime := makeLogsOfSameClusterId("differentTime", initialTime.Add(time.Second), numWithinBucket)
+		err = loadLogsIntoElasticsearch(ac, logsOfDifferentTime)
+		if err != nil {
+			t.Error("Failed to load logs into elasticsearch")
+		}
+		err = loadLogsIntoElasticsearch(ac, []model.LogEntry{newLog})
+		if err != nil {
+			t.Error("Failed to load logs into elasticsearch")
+		}
+		buckets := []service.Bucket{2500}
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		err = countService.CountAndUpdateOccurrences(ctx, newLog, buckets)
+		if err != nil {
+			t.Errorf("Failed to count occurrences: %v", err)
+		}
+		err = countService.CountAndUpdateOccurrences(ctx, newLog, buckets)
```

Comment:
Got lazy here and just ran the function twice, let me just insert directly into the DB to make it isolated